Rewrite the description of WitnessArgs (#642)
nervosnetwork/docs.nervos.org: website/docs/tech-explanation/witness.md

The **order** of WitnessArgs items in the `witnesses` field is also important, as it corresponds to the index in the virtual array created by [script grouping](/docs/tech-explanation/script-group-exe).

#### Example and Test

To illustrate this mechanism, here is a Script that does the following:

1. Prints the contents of its Args (UTF-8 string in this example).
2. Uses `load_script_hash` to get its own ScriptHash.
3. Uses `load_cell_lock_hash` and `load_cell_type_hash` to inspect other Cells’ Lock/Type scripts and print their indices.
4. Uses `load_witness_args` to obtain all GroupInput and GroupOutput data and print them (Witness values stored as UTF-8 strings).

The test transaction is constructed as follows:

```yaml
Inputs:
  0:
    Lock:
      Code Hash: lock_script_1
    Type:
      Code Hash: type_script_1
  1:
    Lock:
      Code Hash: lock_script_1
    Type:
      Code Hash: type_script_2
Outputs:
  0:
    Lock:
      Code Hash: lock_script_1
    Type:
      Code Hash: type_script_2
  1:
    Lock:
      Code Hash: lock_script_1
    Type:
      Code Hash: type_script_1
```

The groups formed are:

- **lock_script_1**: inputs\[0], inputs\[1]

> - **type_script_1**: inputs\[1], outputs\[0]
> - **type_script_2**: inputs\[0], outputs\[1]

Reviewer: I mean the description here seems wrong to me, shouldn't it be:
```suggestion
- **type_script_1**: inputs\[0], outputs\[1]
- **type_script_2**: inputs\[1], outputs\[0]
```

By convention, a group uses the witness of its first input (GroupInput[0]).

- lock_script_1 → first input = 0 → uses witnesses[0]
- type_script_1 → first input = 0 → uses witnesses[0]
- type_script_2 → first input = 1 → uses witnesses[1]

Putting it together:

| Script group (label)     | Entry Witness  | Fields it reads             | Why                                        |
| ------------------------ | -------------- | --------------------------- | ------------------------------------------ |
| lock_script_1 (`lock 1`) | `witnesses[0]` | `lock`                      | Lock Scripts only check inputs             |

> | type_script_2 (`type 2`) | `witnesses[0]` | `input_type`, `output_type` | Type Scripts check both input0 and output1 |
> | type_script_1 (`type 1`) | `witnesses[1]` | `input_type`, `output_type` | Type Scripts check both input1 and output0 |

Reviewer: To my understanding, it should be like the following, correct?
```suggestion
| type_script_1 (`type 1`) | `witnesses[0]` | `input_type`, `output_type` | Type Scripts check both input0 and output1 |
| type_script_2 (`type 2`) | `witnesses[1]` | `input_type`, `output_type` | Type Scripts check both input1 and output0 |
```

<img src={"/img/tech_explanation/witnessArg-structure.png"} width={688} height={353} alt="Structure of WitnessArg" />

This structure is a `table` defined in [Molecule](https://github.com/nervosnetwork/molecule). All three fields are optional (`Opt`), and can be chosen depending on the contract’s requirements.

> - A **Lock Script** executes only when the Cell is an **input**.
> - A **Type Script** executes for both **input** and **output** Cells.
> 
> Therefore, WitnessArgs provides three optional fields to carry proof data for these cases.
> 
> Under this convention, all contracts must read their Witness data through WitnessArgs.  
> The **order** of WitnessArgs items in the `witnesses` field is also important, as it corresponds to the index in the virtual array created by [script grouping](/docs/tech-explanation/script-group-exe).

Reviewer 2: ```suggestion
On CKB, different Scripts run at different times:
- A [Lock Script](/docs/tech-explanation/lock-script) only runs on input Cells (when they are being spent).
- A [Type Script](/docs/tech-explanation/type-script) runs on both input and output Cells (when they are being spent and created)
Because of this, the WitnessArgs structure has three optional fields:
- `lock`: data used by Lock Script
- `input_type`:  data used by Type Script when validating inputs
- "output_type": data used by Type Script when validating outputs
By this convention, all Scripts read their Witness data from WitnessArgs.
The **order** of WitnessArgs items in the `witnesses` field is also important, as it corresponds to the index in the virtual array created by [script grouping](/docs/tech-explanation/script-group-exe).
```
